documentation/contributing/code.markdown

```diff
@@ -28,6 +28,12 @@ In addition MoveIt! has some extra style preferences:
 
  - The ROS logging functionality is utilized and namespacing your logs are recommended, i.e. ``ROS_INFO_NAMED("planning_scene", "Starting listener")``
 
-## MoveIt-Specific
+## Shared Ptrs
 
  - For creating ``shared_ptr`` of any object, use MoveIt!'s standard [macro](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_core/macros/include/moveit/macros/class_forward.h) ``MOVEIT_CLASS_FORWARD(ClassName)`` before the class declaration, and add the include ``#include <moveit/macros/class_forward.h>``. This will create two typedefs of shared pointers - ``<ClassName>Ptr`` and ``<ClassName>ConstPtr`` using either ``boost`` or ``std``.
+
+## CMakeLists
+
+MoveIt! uses *one* folder for all headers from all of its modules: ``include/moveit``. To achieve this, install all headers to `${CATKIN_GLOBAL_INCLUDE_DESTINATION}/moveit` rather than using the normal `CATKIN_PACKAGE_INCLUDE_DESTINATION`.
```

Comment:
This is slightly misleading, because most install statements will look like this:
```
install(DIRECTORY include/ DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION})
```
this installs the `include/moveit` folder that has to be present in package to `<installspace>/include/moveit` without ever mentioning the `moveit` folder.